cli/src/maintenance.rs

```diff
@@ -1356,6 +1375,37 @@ impl SolidoState {
             Some(self_slice_start_slot)
         }
     }
+
+    /// Return None if we observe we moved past `1 -
+    /// SolidoState::END_OF_EPOCH_THRESHOLD`%. Return Some(()) if the above
+    /// condition fails or `self.stake_unstake_any_time` is set to
+    /// `true`.
+    pub fn should_perform_maintenance_at_end_of_epoch(&self) -> Option<()> {
+        if self.stake_unstake_any_time {
+            return Some(());
+        }
+        // Get the slot that the current epoch started.
+        let slots_epoch_begin = self.epoch_schedule.slots_per_epoch * self.clock.epoch;
```

Comment:
This is not the right computation, due to some special behavior near genesis. There is `EpochSchedule::get_first_slot_in_epoch` for this instead.